### PR TITLE
[usm] Enable `MapCleaner` on `connection_protocol` map

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Enable `ebpf.MapCleaner` on `connection_protocol` map.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/usm/ebpf_main.go [hunk 1]
```diff
@@ -354,6 +360,30 @@ func (e *ebpfProgram) init(buf bytecode.AssetReader, options manager.Options) er
...
 
+const connProtoTTL = 3 * time.Minute
+const connProtoCleaningInterval = 5 * time.Minute
+
+func (e *ebpfProgram) setupMapCleaner() *ddebpf.MapCleaner {
+	mapCleaner, err := ddebpf.NewMapCleaner(e.connectionProtocolMap, new(netebpf.ConnTuple), new(netebpf.ProtocolStackWrapper))
+	if err != nil {
+		log.Errorf("error creating map cleaner: %s", err)
+		return nil
+	}
```
Comment (guyarb) [inline pkg/network/usm/ebpf_main.go:371]: return the error to the caller and let it decide what to do with it

File: pkg/network/ebpf/c/protocols/classification/shared-tracer-maps.h [hunk 2]
```diff
@@ -7,20 +7,27 @@
...
-    return bpf_map_lookup_elem(&connection_protocol, &normalized_tup);
+    protocol_stack_wrapper_t empty_wrapper = {0};
+    empty_wrapper.updated = bpf_ktime_get_ns();
+    bpf_map_update_with_telemetry(connection_protocol, &normalized_tup, &empty_wrapper, BPF_NOEXIST);
+    wrapper = bpf_map_lookup_elem(&connection_protocol, &normalized_tup);
+    if (!wrapper) {
+        return NULL;
+    }
+
+    return &wrapper->stack;
```
Comment (guyarb) [inline pkg/network/ebpf/c/protocols/classification/shared-tracer-maps.h:30]: `return __get_protocol_stack`

**Review by guyarb — CHANGES_REQUESTED** [review]
CR'ed

**Review by hmahmood — APPROVED** [review]

File: pkg/network/usm/ebpf_main.go (hunk 1 above)
Comment (p-lambert) [reply pkg/network/usm/ebpf_main.go:371]: This is a private method with one caller, but sure I can move the error handling 🤷

File: pkg/network/ebpf/c/protocols/classification/shared-tracer-maps.h (hunk 2 above)
Comment (p-lambert) [reply pkg/network/ebpf/c/protocols/classification/shared-tracer-maps.h:30]: Good catch

File: pkg/network/usm/ebpf_main.go (hunk 1 above)
Comment (guyarb) [reply pkg/network/usm/ebpf_main.go]: the pattern is the problem, of returnning nil objects and that should indicate the error while errors are silently logged

**Review by guyarb — APPROVED** [review]
LGTM, but the CI is failing, I think you should merge main into your branch

**Pull request merged**